pyccel/ast/core.py

```diff
@@ -1153,6 +1176,15 @@ def __init__(
             if not isinstance(i, Interface):
                 raise TypeError('Only a Inteface instance is allowed.')
 
+        if init_func is not None and not isinstance(init_func, FunctionDef):
```

Comment:
I suggest to use this way for these kind of tests
```python
if not instance(init_func, (type(None), FunctionDef)):
    # error
```

Reply:
Fixed fc326586. I don't know why there isn't a way to do this that doesn't involve calling `type`

Reply:
yah, I don't why they didn't provide NoneType to use it directly.